File: MobileApp/MyApplication/MyApplication/app/src/main/java/edu/uncc/homework4/MyInstanceIDListenerService.java
package edu.uncc.homework4;

import android.content.Intent;

import com.google.android.gms.iid.InstanceIDListenerService;

/**
 * Created by Nitin on 11/19/2017.
 */

public class MyInstanceIDListenerService extends InstanceIDListenerService {

    @Override
    public void onTokenRefresh() {
        super.onTokenRefresh();
        Intent intent = new Intent(this, RegistrationIntentService.class);
        startService(intent);
    }
}

File: MobileApp/CareMeApp/NavigationDrawer/GCM/MyInstanceIDListenerService.cs
using Android.Content;
using Android.App;
using Android.Gms.Iid;

namespace edu.uncc.homework4
{
	[Service (Exported = false), IntentFilter (new [] { "com.google.android.gms.iid.InstanceID" })]
	public class MyInstanceIDListenerService: InstanceIDListenerService
	{
		private const string TAG = "MyInstanceIDLS";

		public override void OnTokenRefresh ()
		{
			var intent = new Intent (this, typeof(RegistrationIntentService));
			StartService (intent);
		}
	}
}

